patroni/dcs/__init__.py

```diff
@@ -1517,39 +1518,40 @@ def __get_patroni_cluster(self, path: Optional[str] = None) -> Cluster:
         """
         if path is None:
             path = self.client_path('')
-        cluster = self._load_cluster(path, self._cluster_loader)
+        cluster = self._load_cluster(path, self._postgresql_cluster_loader)
         if TYPE_CHECKING:  # pragma: no cover
             assert isinstance(cluster, Cluster)
         return cluster
 
-    def is_citus_coordinator(self) -> bool:
-        """:class:`Cluster` instance has a Citus Coordinator group ID.
+    def is_mpp_coordinator(self) -> bool:
+        """:class:`Cluster` instance has a Coordinator group ID.
 
         :returns: ``True`` if the given node is running as the MPP Coordinator.
         """
         return self._mpp.is_coordinator()
 
-    def get_citus_coordinator(self) -> Optional[Cluster]:
-        """Load the Patroni cluster for the Citus Coordinator.
+    def get_mpp_coordinator(self) -> Optional[Cluster]:
+        """Load the PostgreSQL cluster for the MPP Coordinator.
 
           .. note::
-              This method is only executed on the worker nodes (``group!=0``) to find the coordinator.
+              This method is only executed on the worker nodes to find the coordinator.
 
         :returns: Select :class:`Cluster` instance associated with the MPP Coordinator group ID.
         """
         try:
-            return self.__get_patroni_cluster(f'{self._base_path}/{self._mpp.coordinator_group_id}/')
+            return self.__get_postgresql_cluster(f'{self._base_path}/{self._mpp.coordinator_group_id}/')
         except Exception as e:
-            logger.error('Failed to load Citus coordinator cluster from %s: %r', self.__class__.__name__, e)
+            logger.error('Failed to load %s coordinator cluster from %s: %r' %
```

Comment:
@CyberDem0n I'm confused when adding another %s to the error str, if I just add a param seperated by the comma, the error string won't rendered as expected, so I add a % with parentheses(Chatgpt recomment using format or f-strings, but I'm not sure if there will be some compatibility problems). Can you share some experiences?

Reply:
What issue are you getting?
The way it was before (only tuple of args instead of `%` formatting) is the usual way of using `logging.error` and similar functions.

Reply:
Here is what I got when using the way before:
<img width="1791" alt="image" src="https://github.com/zalando/patroni/assets/12639404/b9684d6c-332c-450b-b9c8-a6f673b64f26">

The placeholders seems not rendered, which is really confusing to me.

Reply:
When you use the `%` string it formats the string before calling the mocked function (`logger.error`). In that case you are able to assert against the expected string (`Failed to load Null coordinator cluster from ZooKeeper`) as the first argument, because that is indeed the first argument (the output of a formatted string).

If you call the `logger.error` with arguments and delegate the formatting task to the `logging` module, then you need to adjust the mock assertions accordingly. For example:

```diff
diff --git a/patroni/dcs/__init__.py b/patroni/dcs/__init__.py
index da2d8dcc..481a8e6c 100644
--- a/patroni/dcs/__init__.py
+++ b/patroni/dcs/__init__.py
@@ -1540,8 +1540,8 @@ class AbstractDCS(abc.ABC):
         try:
             return self.__get_postgresql_cluster(f'{self._base_path}/{self._mpp.coordinator_group_id}/')
         except Exception as e:
-            logger.error('Failed to load %s coordinator cluster from %s: %r' %
-                         (self._mpp.type, self.__class__.__name__, e))
+            logger.error('Failed to load %s coordinator cluster from %s: %r',
+                         self._mpp.type, self.__class__.__name__, e)
         return None

     def _get_mpp_cluster(self) -> Cluster:
diff --git a/tests/test_zookeeper.py b/tests/test_zookeeper.py
index 797330d5..301925d0 100644
--- a/tests/test_zookeeper.py
+++ b/tests/test_zookeeper.py
@@ -191,7 +191,10 @@ class TestZooKeeper(unittest.TestCase):
         with patch.object(ZooKeeper, '_postgresql_cluster_loader', Mock(side_effect=Exception)):
             self.assertIsNone(self.zk.get_mpp_coordinator())
             mock_logger.assert_called_once()
-            self.assertIn('Failed to load Null coordinator cluster from ZooKeeper', mock_logger.call_args[0][0])
+            self.assertEqual(mock_logger.call_args[0][0], 'Failed to load %s coordinator cluster from %s: %r')
+            self.assertEqual(mock_logger.call_args[0][1], 'Null')
+            self.assertEqual(mock_logger.call_args[0][2], 'ZooKeeper')
+            self.assertIsInstance(mock_logger.call_args[0][3], ZooKeeperError)

     @patch('patroni.dcs.logger.error')
     def test_get_citus_coordinator(self, mock_logger):
```

I mean, with your approach you are formatting the string before calling the `logger.error`, and with the previous approach the code was delegating the formatting to `logging` module, so it would occur inside of `logging` module.

I suggest to use the approach that delegates to `logging`, which from what I can see is used through the Patroni code base more often -- and actually what the `logging` module suggests in their examples.

Reply:
The same thought applies to other similar changes in the PR.